browser/model/jbosseap.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+let path = require('path');
+let fs = require('fs-extra');
+var rimraf = require('rimraf');
+
+import JbosseapAutoInstallGenerator from './jbosseap-autoinstall';
+import InstallableItem from './installable-item';
+import Installer from './helpers/installer';
+import Logger from '../services/logger';
+import JdkInstall from './jdk-install';
+import Platform from '../services/platform';
+
+class JbosseapInstall extends InstallableItem {
+  constructor(installerDataSvc, targetFolderName, downloadUrl, fileName, jbosseapSha256) {
+    super(JbosseapInstall.KEY, downloadUrl, fileName, targetFolderName, installerDataSvc, true);
+
+    this.sha256 = jbosseapSha256;
+    this.installConfigFile = path.join(this.installerDataSvc.tempDir(), 'jbosseap-autoinstall.xml');
+    this.ConfigFile = path.join(this.installerDataSvc.tempDir(), 'jbosseap-autoinstall.xml.variables');
+    this.runtimeproperties = path.join(this.installerDataSvc.installRoot, '\\developer-studio\\studio', 'runtime_locations.properties');
+    this.addOption('install', this.version, '', true);
+  }
+
+  static get KEY() {
+    return 'jbosseap';
+  }
+
+  installAfterRequirements(progress, success, failure) {
+    progress.setStatus('Installing');
+    this.installGenerator = new JbosseapAutoInstallGenerator(this.installerDataSvc.jbosseapDir(), this.installerDataSvc.jdkDir(), this.version);
+    let installer = new Installer(JbosseapInstall.KEY, progress, success, failure);
+
+    if(fs.existsSync(this.installerDataSvc.jbosseapDir())) {
+      rimraf.sync(this.installerDataSvc.jbosseapDir());
+    }
+
+    Logger.info(JbosseapInstall.KEY + ' - Generate jbosseap auto install file content');
+    let data = this.installGenerator.fileContent();
+    Logger.info(JbosseapInstall.KEY + ' - Generate jbosseap auto install file content SUCCESS');
+    return installer.writeFile(this.installConfigFile, data)
+      .then((result) => {
+        installer.writeFile(this.ConfigFile, 'adminPassword=changeit');
+        return this.postJDKInstall(installer, result);
+      })
+      .then(() => {
+        if(fs.existsSync(this.installerDataSvc.devstudioDir())) {
+          fs.appendFile(this.runtimeproperties , '\njbosseap=' + this.installerDataSvc.jbosseapDir() + ',true');
```

Comment:
@jrichter1 this sounds right to me. Properties file should be modified from jbeap installer on event wwhen devstudio installation is finished. There should be check if devstudio is selected for install. If it is selected then add jbeap install to properties file if present, if not then wait for event about devstudio install is finished and then add jbeap installation to properties file.